src/IO/Importers/ReadSpecThirdOrderPiecewisePolynomial.hpp

```diff
@@ -59,147 +63,153 @@ namespace Actions {
+        // Loop over the remaining times, updating the function of time
+        DataVector highest_derivative(number_of_components);
+        double time_last_updated = start_time;
+        for (size_t row = 1; row < dat_data.rows(); ++row) {
+          // If time of last update has changed, then update the FunctionOfTime
+          // The time of last update is stored in column 1 in the dat file
+          if (dat_data(row, 1) > time_last_updated) {
+            time_last_updated = dat_data(row, 1);
+            for (size_t a = 0; a < number_of_components; ++a) {
+              highest_derivative[a] =
+                  dat_data(row, 5 + (max_deriv + 1) * a + max_deriv);
             }
-            *piecewise_polynomial = spec_functions_of_time[spectre_name];
+            piecewise_polynomial.update(time_last_updated, highest_derivative);
+          } else {
+            file.reset();
+            ERROR("Non-monotonic time found in FunctionOfTime data. "
+                  << "Time " << dat_data(row, 1) << " follows time "
+                  << time_last_updated << " while reading " << spectre_name
+                  << "\n");
           }
-        });
-    return std::forward_as_tuple(std::move(box));
+        }
+        spec_functions_of_time[spectre_name] = piecewise_polynomial;
+      }
+
+      // Loop over elements, and broadcast the functions of time to each
+      // element's DataBox
+      for (auto& element_and_name : get<Tags::RegisteredElements>(box)) {
+        const CkArrayIndex& raw_element_index =
+            element_and_name.first.array_index();
+        // Check if the parallel component of the registered element matches the
+        // callback, because it's possible that elements from other components
+        // with the same index are also registered.
+        // Since the way the component is encoded in `ArrayComponentId` is
+        // private to that class, we construct one and compare.
+        if (element_and_name.first !=
+            observers::ArrayComponentId(
+                std::add_pointer_t<CallbackComponent>{nullptr},
+                raw_element_index)) {
+          continue;
+        }
+
+        // Make an object to hold domain::Tags::FunctionsOfTime data
+        std::unordered_map<
+            std::string,
+            std::unique_ptr<domain::FunctionsOfTime::FunctionOfTime>>
+            spec_functions_of_time_for_tuple;
+        for (const auto& key_and_poly : spec_functions_of_time) {
+          spec_functions_of_time_for_tuple[key_and_poly.first] = static_cast<
+              std::unique_ptr<domain::FunctionsOfTime::FunctionOfTime>>(
+              std::make_unique<domain::FunctionsOfTime::PiecewisePolynomial<3>>(
+                  key_and_poly.second));
+        }
+
+        // Package the domain::Tags::FunctionsOfTime in a TaggedTuple
+        tuples::tagged_tuple_from_typelist<
+            tmpl::list<::domain::Tags::FunctionsOfTime>>
```

Comment:
`tuples::TaggedTuple<::domain::Tags::FunctionsOfTime>` (No need to get a typelist involved.)

Reply:
Why is this passed as a single-element tuple?  Why not just pass the data directly?

Reply:
It's passed as a single-element tuple because the callback action that broadcasts the data to the elements, SetData, takes a tagged tuple (SetData is not always used to broadcast a single entity, as I'm using it here)